app/Models/Bible/Bible.php

```diff
@@ -300,47 +300,50 @@ public function videos()
+        }
+        if ($type_filters['size']) {
+            $query->where('bible_filesets.set_size_code', '=', $type_filters['size']);
+        }
+        if ($type_filters['size_exclude']) {
+            $query->where('bible_filesets.set_size_code', '!=', $type_filters['size_exclude']);
+        }
+    }
+
     public function scopeWithRequiredFilesets($query, $type_filters)
-    {   
+    {
         $permitted_filesets = $type_filters['access_control']->identifiers;
 
         if ($type_filters['tag_exclude']) {
-            $opus_filesets = BibleFilesetTag::select('hash_id')->where('description', $type_filters['tag_exclude'])->pluck('hash_id')->toArray();
+            $opus_filesets = BibleFilesetTag::select('hash_id')
+                ->where('description', $type_filters['tag_exclude'])
+                ->pluck('hash_id')
+                ->toArray();
             $permitted_filesets = array_diff($type_filters['access_control']->identifiers, $opus_filesets);
         }
 
         $queryIn = sprintf("bible_filesets.hash_id IN ('" . implode("', '", $permitted_filesets) . "')");
+
         return $query->whereHas('filesets', function ($q) use ($type_filters, $queryIn) {
             $q->whereRaw($queryIn);
             if ($type_filters['media']) {
                 $q->where('bible_filesets.set_type_code', $type_filters['media']);
             }
-            if ($type_filters['media_exclude']) {
-                $q->where('bible_filesets.set_type_code', '!=', $type_filters['media_exclude']);
-            }
-            if ($type_filters['size']) {
-                $q->where('bible_filesets.set_size_code', '=', $type_filters['size']);
-            }
-            if ($type_filters['size_exclude']) {
-                $q->where('bible_filesets.set_size_code', '!=', $type_filters['size_exclude']);
-            }
+            $this->setConditionFilesets($q, $type_filters);
         })->with(['filesets' => function ($q) use ($type_filters, $queryIn) {
             $q->with(['meta' => function ($subQuery) {
-              $subQuery->where('admin_only', 0);
+                $subQuery->where('admin_only', 0);
             }]);
-            $q->whereRaw($queryIn);
-            if ($type_filters['media']) {
-                $q->where('bible_filesets.set_type_code', $type_filters['media']);
-            }
-            if ($type_filters['media_exclude']) {
-                $q->where('bible_filesets.set_type_code', '!=', $type_filters['media_exclude']);
-            }
-            if ($type_filters['size']) {
-                $q->where('bible_filesets.set_size_code', '=', $type_filters['size']);
-            }
-            if ($type_filters['size_exclude']) {
-                $q->where('bible_filesets.set_size_code', '!=', $type_filters['size_exclude']);
-            }
+            $q->whereRaw(
+                'EXISTS (SELECT 1 FROM bible_filesets bf_tmp WHERE bf_tmp.id = bible_filesets.id and ' . $queryIn . ')'
```

Comment:
I have added the limit 1 to EXISTS subquery and It didn't reduce the number of rows returned, it works. But, I was reading in some posts and they said that is no necessary to add the limit 1 inside a exits subquery becuase "The purpose of EXISTS() is to perform the query only until it can decide if there are any rows in that table".